src/hooks/useLatestFunc.ts

```diff
@@ -1,15 +1,19 @@
 import { useRef, useEffect, useCallback } from 'react';
+import type { Maybe } from '../types';
 
 // https://reactjs.org/docs/hooks-faq.html#what-can-i-do-if-my-effect-dependencies-change-too-often
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-export function useLatestFunc<T extends (...args: any[]) => any>(fn: T) {
+export function useLatestFunc<T extends Maybe<(...args: any[]) => any>>(fn: T): T {
```

Comment:
Without `:T`, the type is 

![image](https://user-images.githubusercontent.com/1452717/189754869-8b971865-4a05-4dd7-8255-1490bdcbaa33.png)